### PR TITLE
describe_cat_var function and its test. Also contains some changes to helper function

### LINKED ISSUE #26: Implement describe_cat_vars function

### LINKED ISSUE #29: code review for describe_cat_var

### LINKED ISSUE #31: Create test function for decribe_cat_vars

### PR DESCRIPTION
**Pull request by sgauravm**
FIxes #26 #31

### CODE REVIEW

File: R/eda.R [hunk 1]
```diff
@@ -116,11 +116,42 @@ describe_num_var <- function(dataframe, num_vars) {
...
+
+  # Check the input cat_vars is a vector of column names of dataframe
+  if (!all(cat_vars %in% colnames(dataframe))) {
+    stop("The argument 'cat_vars' should be a subset of the column names of the dataframe.")
+  }
+
+  # Select the categorical variables to work with
+  df <- dataframe %>%
+    select(cat_vars)
+
```
Comment (marvinmin) [inline R/eda.R:140]: I'm not sure if there is a way to check only categorical columns are selected. Similarly to the python function, if the numeric columns are selected, the plot does not look good.

File: R/eda.R [hunk 2]
```diff
@@ -116,11 +116,42 @@ describe_num_var <- function(dataframe, num_vars) {
 #' describe_num_var(X, cat_vars)
 #'
 describe_cat_var <- function(dataframe, cat_vars) {
-  #TODO implement function
-  print(cat_vars)
+  # Check the input dataframe is a dataframe
+  if (!is.data.frame(dataframe)) {
+    stop("The value of the argument 'dataframe' should be of type  'data.frame' or 'tibble'.")
```
Comment (marvinmin) [inline R/eda.R]:
```suggestion
    stop("The value of the argument 'dataframe' should be of type 'data.frame' or 'tibble'.")
```
There is an extra space.

File: tests/testthat/test-eda.R [hunk 3]
```diff
@@ -56,3 +56,41 @@ test_describe_num_var <- function() {
...
+  })
+
+  test_that("The plot should be a bar chart and without y mapping.", {
+    expect_true("GeomBar" %in% c(class(result$layers[[1]]$geom)))
+    expect_true(is.null(rlang::get_expr(result$mapping$y)))
+  })
+
+  test_that("Corresponding error message should be expected if the dataframe argument is not a dataframe.", {
+    expect_error(describe_cat_var("abc", cat_var),
+                 regexp = "The value of the argument 'dataframe' should be of type  'data.frame' or 'tibble'.")
```
Comment (marvinmin) [inline tests/testthat/test-eda.R]:
```suggestion
                 regexp = "The value of the argument 'dataframe' should be of type 'data.frame' or 'tibble'.")
```
Change accordingly

**Review by marvinmin — CHANGES_REQUESTED** [review]
The R function and its test are great! There are only minor changes needed.

File: R/eda.R (hunk 1 above)
Comment (marvinmin) [reply R/eda.R:140]: I guess for the same reason in the python function, we can ignore this comment.

File: tests/testthat/test-eda.R [hunk 4]
```diff
@@ -56,3 +56,41 @@ test_describe_num_var <- function() {
     regexp = "Only numeric columns expected, please check the input.")
   })
 }
+
+
+test_describe_cat_var <- function() {
```
Comment (marvinmin) [inline tests/testthat/test-eda.R:215]: I think we need to add documentation for the test function as well. It is required in milestone 2.

Comment (sgauravm) [reply tests/testthat/test-eda.R:215]: added documentation

**Review by marvinmin — APPROVED** [review]
Thanks for making the changes!

**Pull request merged**